sphinx.go

```diff
@@ -660,8 +674,16 @@ func unwrapPacket(onionPkt *OnionPacket, sharedSecret *Hash256,
 	// With the MAC checked, and the payload decrypted, we can now parse
 	// out the payload so we can derive the specified forwarding
 	// instructions.
-	var hopPayload HopPayload
-	if err := hopPayload.Decode(bytes.NewReader(hopInfo[:])); err != nil {
+	hopPayload := HopPayload{}
+	if tlvPayloadOnly {
```

Comment:
rather create a constructor that takes `tlvPayloadOnly` as a param? 
that way we also explicitly set the Legacy type. We should not depend on the fact that it happens to be the zero value